Fromdiff --git a/third_party/libxml/src/parser.c b/third_party/libxml/src/parser.c
index 33786f08354d..745c3da1c72e 100644
--- a/parser.c
+++ b/parser.c
@@ -3426,7 +3426,7 @@ xmlParseNameComplex(xmlParserCtxtPtr ctxt) {
         xmlFatalErr(ctxt, XML_ERR_NAME_TOO_LONG, "Name");
         return(NULL);
     }
-    if ((*ctxt->input->cur == '\n') && (ctxt->input->cur[-1] == '\r')) {
+    if (ctxt->input->cur > ctxt->input->base && (*ctxt->input->cur == '\n') && (ctxt->input->cur[-1] == '\r')) {
         if (ctxt->input->base > ctxt->input->cur - (len + 1)) {
             return(NULL);
         }
-- 
2.12.2.715.g7642488e1d-goog

